From 43d1900ba9016dc8a6be9d3dded31497444f725d Mon Sep 17 00:00:00 2001
From: Will Ruggiano <wmruggiano@gmail.com>
Date: Fri, 13 Dec 2024 11:11:38 -0800
Subject: [PATCH] fix: use package.cpath

---
 lua/ivy/libivy.lua | 22 +---------------------
 1 file changed, 1 insertion(+), 21 deletions(-)

diff --git a/lua/ivy/libivy.lua b/lua/ivy/libivy.lua
index cb77ce6..ab8bca9 100644
--- a/lua/ivy/libivy.lua
+++ b/lua/ivy/libivy.lua
@@ -1,24 +1,4 @@
-local library_path = (function()
-  local root = string.sub(debug.getinfo(1).source, 2, #"/libivy.lua" * -1)
-  local release_path = root .. "../../target/release"
-  local current_vim_version = vim.version()
-  local minimum_supported_version = vim.version.parse "0.9.5"
-
-  local is_windows
-
-  if vim.version.gt(current_vim_version, minimum_supported_version) then
-    is_windows = vim.uv.os_uname().sysname == "Windows_NT"
-  else
-    is_windows = vim.loop.os_uname().sysname == "Windows_NT"
-  end
-
-  if is_windows then
-    return package.searchpath("ivyrs", release_path .. "/?.dll;")
-  else
-    return package.searchpath("libivyrs", release_path .. "/?.so;" .. release_path .. "/?.dylib;")
-  end
-end)()
-
+local library_path = package.searchpath("libivyrs", package.cpath);
 local ffi = require "ffi"
 local ok, ivy_c = pcall(ffi.load, library_path)
 if not ok then
